### PR TITLE
feat: Add Swap Kit components className override.

### PR DESCRIPTION
**Pull request by cpcramer**
**What changed? Why?**

Add className override to `SwapButton`, `SwapMessage`, `SwapToggleButton`, `Swap`, `QuoteWarning` , and `SwapAmountInput`.

Add missing React input types for the respective components.

**Notes to reviewers**

**How has it been tested?**
Tested locally, by adding a background color using the className field:

Screenshot 2024-06-25 at 4 14 51 PM

### CODE REVIEW

File: src/swap/components/SwapMessage.tsx [hunk 1]
```diff
@@ -25,7 +26,10 @@ export function SwapMessage() {
 
   return (
     <div className="flex pt-2">
-      <span className={text.label2} data-testid="ockSwapMessage_Message">
+      <span
+        className={cn(text.label2, className)}
+        data-testid="ockSwapMessage_Message"
+      >
```
Comment (cpcramer) [inline src/swap/components/SwapMessage.tsx:32]: Left as a Span element. Can update to a div if preferred!

Comment (kyhyco) [reply src/swap/components/SwapMessage.tsx:32]: nit: let's combine the `div` and `span` into a single element

File: src/swap/types.ts [hunk 2]
```diff
@@ -102,6 +116,7 @@ export type SwapAmountInputReact = {
   swappableTokens?: Token[]; // Swappable tokens
   token?: Token; // Selected token
   type: 'to' | 'from'; // Identifies if component is for toToken or fromToken
+  className?: string; // Optional className override for top div element.
```
Comment (kyhyco) [inline src/swap/types.ts:119]: are we overriding `QuoteWarning`?

Comment (cpcramer) [reply src/swap/types.ts:119]: No. Will update now!

Comment (cpcramer) [reply src/swap/types.ts:120]: Overriding here https://github.com/coinbase/onchainkit/pull/719/files#diff-9bca97e90be8d6cfaa2c6813dbf1d684ed792b80b395f6b033d3d06963f23371R99

File: src/swap/components/SwapMessage.tsx (hunk 1 above)
Comment (cpcramer) [reply src/swap/components/SwapMessage.tsx]: Updated

File: src/swap/types.ts (hunk 2 above)
Comment (kyhyco) [reply src/swap/types.ts:120]: Are we sure we want to add className to QuoteWarning and RawTransactionData?

Comment (cpcramer) [reply src/swap/types.ts:120]: Hmm I don't think QuoteWarning and RawTransactionData overriding is very important. I don't think we need it unless users start to ask for it.

Comment (kyhyco) [reply src/swap/types.ts:120]: derp the code folding... see it now lol

We don't need `className` override to `QuoteWarning` right?

Comment (cpcramer) [reply src/swap/types.ts:119]: Nope! Removed

File: site/docs/components/App.tsx [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
+import { OnchainKitProvider } from '../../../src';
 import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
 import { WagmiProvider, createConfig, http } from 'wagmi';
 import { base } from 'wagmi/chains';
 import { coinbaseWallet } from 'wagmi/connectors';
 
-import '@coinbase/onchainkit/styles.css';
-// import '../../../src/styles.css';
+// import '@coinbase/onchainkit/styles.css';
+import '../../../src/styles.css';
```
Comment (kyhyco) [inline site/docs/components/App.tsx:11]: revert this file change

File: site/docs/pages/swap/swap.mdx [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import { ConnectAccount } from '@coinbase/onchainkit/wallet';
-import { Swap, SwapAmountInput, SwapButton, SwapMessage, SwapToggleButton } from '@coinbase/onchainkit/swap';
+import { Swap, SwapAmountInput, SwapButton, SwapMessage, SwapToggleButton } from '../../../../src/swap';
```
Comment (kyhyco) [inline site/docs/pages/swap/swap.mdx:2]: include this doc in a separate PR after we cut a new release

File: site/docs/styles.css [hunk 5]
```diff
@@ -43,8 +51,8 @@
   top: 50%;
   left: 50%;
   transform: translate(-50%, 0%);
-  width: 500px;
-  height: 500px;
+  max-width: 500px;
+  max-height: 500px;
```
Comment (kyhyco) [inline site/docs/styles.css:55]: Do we need this in this PR?

**Comment by Zizzamia** [comment]
@cpcramer you made several coding changes in this PR, where are the tests to help cover those changes?

File: src/swap/types.ts [hunk 6]
```diff
@@ -78,6 +85,13 @@ export type GetSwapMessageParams = {
   toToken?: Token;
 };
 
+/**
+ * Note: exported as public Type
```
Comment (Zizzamia) [inline src/swap/types.ts:89]: @cpcramer it says `exported as public Type`, but in this PR is not either exported or documented.

That comment has a deep meaning, which is not reflected in the code you wrote.

If you add that comment, than you want to make sure to export and document the type in `sitet`.

Comment (cpcramer) [reply src/swap/types.ts:89]: Got it. Will fix!

File: src/swap/components/SwapMessage.tsx [hunk 7]
```diff
@@ -24,10 +25,11 @@ export function SwapMessage() {
   });
 
   return (
-    <div className="flex pt-2">
-      <span className={text.label2} data-testid="ockSwapMessage_Message">
-        {message}
-      </span>
+    <div
+      className={cn('flex pt-2', text.label2, className)}
+      data-testid="ockSwapMessage_Message"
```
Comment (cpcramer) [inline src/swap/components/SwapMessage.tsx:30]: Perhaps `ockSwapMessage_Container` is a better name here?

File: src/swap/components/SwapButton.tsx [hunk 8]
```diff
@@ -61,19 +65,23 @@ export function SwapButton({ disabled = false, onSubmit }: SwapButtonReact) {
...
         background.primary,
         'w-full rounded-xl',
         'mt-4 px-4 py-3 font-medium text-base text-white leading-6',
-        isDisabled && !swapLoadingState?.isSwapLoading && pressable.disabled,
+        isDisabled && !isLoading && pressable.disabled,
+        className,
       )}
       onClick={handleSubmit}
       disabled={isDisabled}
+      data-testid="ockSwapButton_Button"
```
Comment (cpcramer) [inline src/swap/components/SwapButton.tsx:82]: `ockSwapButton_Container` here?

Comment (kyhyco) [reply src/swap/components/SwapButton.tsx:82]: `ockSwapButton_Button` is good

File: src/swap/components/SwapButton.tsx [hunk 9]
```diff
@@ -61,19 +65,23 @@ export function SwapButton({ disabled = false, onSubmit }: SwapButtonReact) {
...
     >
-      {swapLoadingState?.isSwapLoading ? (
+      {!isLoading ? (
```
Comment (kyhyco) [inline src/swap/components/SwapButton.tsx:84]: nice!

File: src/swap/components/SwapButton.tsx (hunk 8 above)
Comment (cpcramer) [reply src/swap/components/SwapButton.tsx:82]: sweeet!

**Review by kyhyco — APPROVED** [review]

**Pull request merged**